src/game/common/system/mempoolobj.h

```diff
@@ -104,10 +104,11 @@ public: \
 #define IMPLEMENT_POOL(classname) IMPLEMENT_NAMED_POOL(classname, classname);
 
 #define IMPLEMENT_ABSTRACT_POOL(classname) \
-private: \
+protected: \
```

Comment:
whats the reason for this?
getObjectMemoryPool is confirmed as private by a linker map left in Generals Demo, and others we chose to make private

Reply:
Well, it is no valid C++. A delete operator needs to be visible to the compiler.
It results in errors like these: `error: ‘static void Thing::operator delete(void*)’ is private within this context`

The only reason this compiles under WIndows is probably MSVC being a non-compliant C++ compiler :)

Reply:
@feliwir Where did the error originate from?

Reply:
My understanding is that it's perfectly legal to have a private operator be overloaded to public in a derived. I even made a minimal test case and gcc/clang appears to compile it fine. I'm just scared that this change will lead to cases where previously there was a compiler error and now there is only a runtime error.
https://godbolt.org/z/Wz15edWfd

Reply:
@duncanspumpkin  you got the error wrong. The operator is not overloaded by the derived class at all. Resulting in exactly the error i got: https://godbolt.org/z/bGvabq91K

Reply:
But no derived should exist that don't have a non abstract pool macro unless they are also an abstract pool. So this does suggest that there are other issues. Can you list all the types it errors on. UpdateModule for example I don't think should ever be created except through a derived.

Reply:
Fwiw msvc also errors on hidden private in godbolt in your test so i think it's a problem in some ifdef code.

Reply:
Yes, in theory there would be some place where the delete operator is called (which would not be allowed), however GCC does not show that place. I'm trying to compile with `clang` to get a better error description, however there's another chain of issues when trying to compile with clang

Reply:
Could you test if that linux error still happens with this change
* #537

Reply:
@xezon  it doesn't work with these changes. I isolated the issue down to the virtual destructors:
https://godbolt.org/z/dsMrerv78

This compiles for MSVC, but not for GCC/Clang